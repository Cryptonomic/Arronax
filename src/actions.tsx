import { TezosFilter } from 'conseiljs';
import { DataView } from 'types';
import * as constants from './constants';

export interface SwitchTab {
  type: constants.SWITCH_TAB;
  dataView: DataView;
}

export interface SetFilter {
  type: constants.SET_FILTER;
  filters: TezosFilter;
}

export interface SetNetwork {
  type: constants.SET_NETWORK;
  network: string;
}

export type ArronaxAction = SwitchTab | SetFilter | SetNetwork;

export const switchTab = (dataView: DataView): SwitchTab =>
  ({
    dataView,
    type: constants.SWITCH_TAB
  });

export const setFilter = (filters: TezosFilter): SetFilter =>
  ({
    filters,
    type: constants.SET_FILTER
  });

export const setNetwork = (network: string): SetNetwork =>
  ({
    network,
    type: constants.SET_NETWORK
  });
