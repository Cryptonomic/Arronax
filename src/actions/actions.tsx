import * as constants from '../constants/constants';

export interface SwitchMode {
    type: constants.SWITCH_MODE;
}

export interface SwitchTab {
    type: constants.SWITCH_TAB;
}

export interface SetFilter {
    type: constants.SET_FILTER;
}

export interface ResetAll {
    type: constants.RESET_ALL;
}

export type ArronaxAction = SwitchMode | SwitchTab | SetFilter | ResetAll;

export function switchMode(): SwitchMode {
    return {
        type: constants.SWITCH_MODE
    };
}

export function switchTab(): SwitchTab {
    return {
        type: constants.SWITCH_TAB
    };
}

export function setFilter(): SetFilter {
    return {
        type: constants.SET_FILTER
    };
}

export function resetAll(): ResetAll {
    return {
        type: constants.RESET_ALL
    };
}
