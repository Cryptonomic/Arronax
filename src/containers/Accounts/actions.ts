import { TezosAccount } from 'types';
import { FETCH_ACCOUNTS, SET_ACCOUNTS } from './constants';

export interface FetchAccounts {
  type: FETCH_ACCOUNTS;
}

export interface SetAccounts {
  type: SET_ACCOUNTS;
  accounts: TezosAccount[];
}

export type AccountsAction = FetchAccounts | SetAccounts;

export const fetchAccounts = (): FetchAccounts => ({
  type: FETCH_ACCOUNTS
});

export const setAccounts = (accounts: TezosAccount[]): SetAccounts => ({
  accounts,
  type: SET_ACCOUNTS
});
