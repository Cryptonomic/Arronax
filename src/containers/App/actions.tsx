import { TezosFilter } from 'conseiljs';
import { SET_FILTER } from './constants';

export interface SetFilter {
  type: SET_FILTER;
  filters: TezosFilter;
  network: string;
}

export type ArronaxAction = SetFilter;

export const setFilter = (filters: TezosFilter, network: string): SetFilter => ({
  filters,
  network,
  type: SET_FILTER
});
