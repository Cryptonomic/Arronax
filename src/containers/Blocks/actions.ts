import { TezosBlock } from 'types';
import { FETCH_BLOCKS, SET_BLOCKS } from './constants';

export interface FetchBlocks {
  type: FETCH_BLOCKS;
}

export interface SetBlocks {
  type: SET_BLOCKS;
  blocks: TezosBlock[];
}

export type BlocksAction = FetchBlocks | SetBlocks;

export const fetchBlocks = (): FetchBlocks => ({
  type: FETCH_BLOCKS
});

export const setBlocks = (blocks: TezosBlock[]): SetBlocks => ({
  blocks,
  type: SET_BLOCKS
});
