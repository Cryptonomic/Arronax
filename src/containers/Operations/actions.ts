import { TezosOperation } from 'types';
import {FETCH_OPERATIONS, FETCH_OPERATIONS_FAILED, SET_OPERATIONS} from './constants';

export interface FetchOperations {
  type: FETCH_OPERATIONS;
}

export interface SetOperations {
  type: SET_OPERATIONS;
  operations: TezosOperation[];
}

export interface FetchOperationsFailed {
  type: FETCH_OPERATIONS_FAILED;
  error: string;
}

export type OperationsAction = FetchOperations | SetOperations | FetchOperationsFailed;

export const fetchOperations = (): FetchOperations => ({
  type: FETCH_OPERATIONS
});

export const setOperations = (operations: TezosOperation[]): SetOperations => ({
  operations,
  type: SET_OPERATIONS
});

export const fetchOperationsFailed = (error: string): FetchOperationsFailed => ({
    type: FETCH_OPERATIONS_FAILED,
    error
});
