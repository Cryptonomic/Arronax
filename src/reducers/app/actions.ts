import {
    SET_ITEMS,
    SET_FILTER,
    SET_TAB,
    SET_LOADING,
    SET_COLUMNS,
    SET_ATTRIBUTES,
    ADD_FILTER,
    REMOVE_FILTER,
    REMOVE_ALL_FILTERS,
    CHANGE_FILTER,
    SET_AVAILABLE_VALUES,
    COMPLETE_FULL_LOAD,
    SET_FILTER_COUNT,
    SET_SORT,
    SET_ENTITIES,
    SET_ENTITY,
    SET_ENTITY_PROPERTIES,
    INIT_FILTER,
    SET_CONFIG,
    ADD_CONFIG,
    REMOVE_CONFIG,
    SET_AGGREGATIONS,
    SET_SUBMIT,
    SET_QUERY_FILTERS,
} from './types';

import { AttributeDefinition, EntityDefinition } from 'conseiljs';
import { Sort, Filter, Config, Aggregation } from '../../types';

// remove?
export function setItemsAction(entity: string, items: any[]) {
    return {
        type: SET_ITEMS,
        entity,
        items,
    };
}

export function setColumnsAction(entity: string, columns: AttributeDefinition[], sorts: Sort[], aggregations: Aggregation[]) {
    return {
        type: SET_COLUMNS,
        entity,
        columns,
        sorts,
        aggregations,
    };
}

// remove?
export function setFilterAction(filters: Filter[]) {
    return {
        type: SET_FILTER,
        filters,
    };
}

export function setTabAction(entity: string) {
    return {
        type: SET_TAB,
        entity,
    };
}

export function setLoadingAction(isLoading: boolean) {
    return {
        type: SET_LOADING,
        isLoading,
    };
}

export function setConfigAction(config: Config) {
    return {
        type: SET_CONFIG,
        config,
    };
}

export function addConfigAction(config: Config, isUse: boolean) {
    return {
        type: ADD_CONFIG,
        config,
        isUse,
    };
}

export function removeConfigAction(index: number) {
    return {
        type: REMOVE_CONFIG,
        index,
    };
}

export function addFilterAction(entity: string) {
    return {
        type: ADD_FILTER,
        entity,
    };
}

export function removeFilterAction(entity: string, index: number) {
    return {
        type: REMOVE_FILTER,
        entity,
        index,
    };
}

export function removeAllFiltersAction(entity: string) {
    return {
        type: REMOVE_ALL_FILTERS,
        entity,
    };
}

export function changeFilterAction(entity: string, filter: Filter, index: number) {
    return {
        type: CHANGE_FILTER,
        entity,
        filter,
        index,
    };
}

export function completeFullLoadAction(isFullLoaded: boolean) {
    return {
        type: COMPLETE_FULL_LOAD,
        isFullLoaded,
    };
}

export function setAvailableValuesAction(entity: string, attribute: string, availableValues: any[]) {
    return {
        type: SET_AVAILABLE_VALUES,
        entity,
        attribute,
        availableValues,
    };
}

// remove?
export function setFilterCountAction(count: number) {
    return {
        type: SET_FILTER_COUNT,
        count,
    };
}

export function setSortAction(entity: string, sorts: Sort[]) {
    return { type: SET_SORT, entity, sorts };
}

export function setEntitiesAction(entities: EntityDefinition[]) {
    return {
        type: SET_ENTITIES,
        entities,
    };
}

export function setEntityAction(entity: string) {
    return {
        type: SET_ENTITY,
        entity,
    };
}

export function setEntityPropertiesAction(entity: string, filters: Filter[], sorts: Sort[], columns: any[], items: any[], aggregations: Aggregation[]) {
    return {
        type: SET_ENTITY_PROPERTIES,
        entity,
        filters,
        sorts,
        columns,
        items,
        aggregations,
    };
}

export function initFilterAction(entity: string, filters: Filter[]) {
    return {
        type: INIT_FILTER,
        entity,
        filters,
    };
}

export function setAttributesAction(attributes: any) {
    return {
        type: SET_ATTRIBUTES,
        attributes,
    };
}

export function setAggregationAction(entity: string, aggregations: Aggregation[], sorts: Sort[]) {
    return {
        type: SET_AGGREGATIONS,
        entity,
        aggregations,
        sorts,
    };
}

export function setSubmitAction(entity: string, items: any[], filterCount: number) {
    return {
        type: SET_SUBMIT,
        entity,
        items,
        filterCount,
    };
}

export const setQueryFilters = (entity: string, queryFilters: any): any => ({
    type: SET_QUERY_FILTERS,
    entity,
    queryFilters,
});
