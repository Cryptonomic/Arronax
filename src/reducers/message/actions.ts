import { CLEAR_MESSAGE_STATE, ADD_MESSAGE } from './types';
import actionCreator from '../../utils/reduxHelpers';

export const clearMessageState = actionCreator(CLEAR_MESSAGE_STATE);
export const createMessage = actionCreator(
  ADD_MESSAGE,
  'message',
  'isError',
  'hash',
  'localeParam'
);
