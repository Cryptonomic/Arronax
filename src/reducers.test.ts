import arronaxReducer from "./reducers";
import * as actions from "./actions";
/**
 * FIXTURES
 */
const testFilter = {
  block_id: [],
  block_level: [],
  block_netid: [],
  block_protocol: [],
  operation_id: [],
  operation_source: [],
  operation_destination: [],
  operation_participant: [],
  operation_kind: [],
  account_id: [],
  account_manager: [],
  account_delegate: [],
  limit: 100,
};

const initialState = arronaxReducer(undefined, {} as any);

/**
 * STORIES
 */
describe("Arronax stories", () => {
  describe("initial state", () => {
    it("should match a snapshot", () => {
      expect(initialState).toMatchSnapshot();
    });
  });
  describe("switch tab", () => {
    it("should switch to random tab", () => {
      const tab: number = Math.floor(Math.random() * 10);
      const action = actions.switchTab(tab);
      const state = arronaxReducer(initialState, action);
      expect(state.dataView).toEqual(tab);
    });
  });
  describe("set network", () => {
    it("should set network", () => {
      const action = actions.setNetwork("someNetwork");
      const state = arronaxReducer(initialState, action);
      expect(state.network).toEqual("someNetwork");
    });
  });
  describe("set filters", () => {
    it("should set filters", () => {
      const action = actions.setFilter(testFilter);
      const state = arronaxReducer(initialState, action);
      expect(state.filters).toEqual(testFilter);
    });
    it("should change filter limit", () => {
      const LIMIT = 10;
      const action = actions.setFilter({ ...testFilter, limit: LIMIT });
      const state = arronaxReducer(initialState, action);
      expect(state.filters.limit).not.toEqual(testFilter.limit);
      expect(state.filters.limit).toEqual(LIMIT);
    });
  });
});
