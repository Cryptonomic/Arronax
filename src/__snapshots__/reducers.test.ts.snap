// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arronax stories initial state should match a snapshot 1`] = `
Object {
  "dataView": 0,
  "filters": Object {
    "account_delegate": Array [],
    "account_id": Array [],
    "account_manager": Array [],
    "block_id": Array [],
    "block_level": Array [],
    "block_netid": Array [],
    "block_protocol": Array [],
    "limit": 100,
    "operation_destination": Array [],
    "operation_id": Array [],
    "operation_kind": Array [],
    "operation_participant": Array [],
    "operation_source": Array [],
  },
  "network": "zeronet",
}
`;
